tsc/src/core/global_basic.hpp

```diff
@@ -98,10 +98,24 @@
 #include <SDL.h>
 #include <SDL_opengl.h>
 #include <SDL_image.h>
-#include <SDL_mixer.h>
 #include <SDL_ttf.h>
 #include <SDL_syswm.h>
 
+// SFML
+/*
+   XXX: This is a hack! X11 defines a macro Status that expands to int, so SFML's
+   Status enum ends up being invalid syntax. Who thought that was a good idea
+   again?
+*/
```

Comment:
X11 had definitions that already clashed with SDL and CEGUI. Take a look at how global_basic.hpp looked like in the old OpenGL/SDL port and you’ll see there are comments about that: https://github.com/Secretchronicles/TSC/blob/devel/tsc/src/core/global_basic.hpp#L87-L89

The comment only talks about CEGUI, but AFAIK it affects SDL as well. Try moving all the SFML includes before the X11 includes (so they are kept as a block of related statements) and see if that resolves the problem. If not, we’ll have to stick to this hack for now.